pysph/base/opencl.py

```diff
@@ -35,6 +35,52 @@ def set_queue(q):
     _queue = q
 
 
+class DeviceArray(cl.array.Array):
+    def __init__(self, dtype, n=0):
+        self.queue = get_queue()
+        self.length = n
+        if n == 0:
+            n = 16
+        self.alloc = n
```

Comment:
Hmm, can you reuse `self.set_data` here to do this and remove some duplicated code?